docs/diagrams/ViewSequenceDiagram.puml

```diff
@@ -0,0 +1,81 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":LinkedoutParser" as LinkedoutParser LOGIC_COLOR
+participant ":ViewCommandParser" as ViewCommandParser LOGIC_COLOR
+participant "d:ViewCommand" as ViewCommand LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "NameContainsAllKeywordsPredicate" as NameContainsAllKeywordsPredicate MODEL_COLOR
+participant ":Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("view Alex Tan")
+activate LogicManager
+
+LogicManager -> LinkedoutParser : parseCommand("view Alex Tan")
+activate LinkedoutParser
+
+create ViewCommandParser
+LinkedoutParser -> ViewCommandParser
+activate ViewCommandParser
+
+ViewCommandParser --> LinkedoutParser
+deactivate ViewCommandParser
+
+LinkedoutParser -> ViewCommandParser : parse(args)
+activate ViewCommandParser
+
+create ViewCommand
+ViewCommandParser -> ViewCommand
+activate ViewCommand
+
+ViewCommand --> ViewCommandParser : d
+deactivate ViewCommand
+
+ViewCommandParser --> LinkedoutParser : d
+deactivate ViewCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+ViewCommandParser -[hidden]-> LinkedoutParser
+destroy ViewCommandParser
+
+LinkedoutParser --> LogicManager : d
+deactivate LinkedoutParser
+
+LogicManager -> ViewCommand : execute()
+activate ViewCommand
+
+create NameContainsAllKeywordsPredicate
```

Comment:
if not mistaken NameContainsAllKeywordsPredicate is created before execute() being called

Reply:
Thanks for the spot, updated!